### PR TITLE
Remove deprecated AntlrSourceVirtualDirectory & AntlrSourceVirtualDirectoryImpl

### PR DESCRIPTION
**Pull request by eskatos**
With Gradle 8.0 you should use the already existing `AntlrSourceDirectorySet` source set extension instead.

See
* https://github.com/gradle/gradle/issues/21790

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test this

**Review by nathan-contino — APPROVED** [review]
Docs LGTM, optional suggestion

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 1]
```diff
@@ -63,6 +63,12 @@ For more information, see <<old_artifact_transforms_api>>.
 
 The legacy `IncrementalTaskInputs` API has been removed. For more information, see <<incremental_task_inputs_deprecation>>. This change also affects Kotlin Gradle Plugin and Android Gradle Plugin. With Gradle 8.0 you should use Kotlin Gradle Plugin 1.6.10 or later and Android Gradle Plugin 7.3.0 with `android.experimental.legacyTransform.forceNonIncremental=true` property or later.
 
+==== Legacy AntlrSourceVirtualDirectory API
+
+The legacy `AntlrSourceVirtualDirectory` API has been removed.
+This change affects the `antlr` plugin.
+With Gradle 8.0 you should use the already existing `AntlrSourceDirectorySet` source set extension instead.
```
[the remark this answers is not in the document]
Comment (eskatos) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:70]: Thanks Nate!

**Review by ljacomet — APPROVED** [review]

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Pull request merged**